### PR TITLE
LPS-139647 Create new LayoutPageTemplateStructure upgrade process

### PR DESCRIPTION
**Pull request by markgulacsy**
Hi @balazssk,
Could you review this upgradeProcess, please?
Thanks!

The LPS -> https://issues.liferay.com/browse/LPS-139647

### CODE REVIEW

File: modules/apps/layout/layout-page-template-service/src/main/java/com/liferay/layout/page/template/internal/upgrade/v3_4_4/LayoutPageTemplateStructureUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+	public LayoutPageTemplateStructureUpgradeProcess(
+		LayoutLocalService layoutLocalService) {
+
+		_layoutLocalService = layoutLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
```
Comment (balazssk) [inline modules/apps/layout/layout-page-template-service/src/main/java/com/liferay/layout/page/template/internal/upgrade/v3_4_4/LayoutPageTemplateStructureUpgradeProcess.java:39]: Layout is not part of layout-page-template-service module, we need to use _layoutLocalService to query it

File: modules/apps/layout/layout-page-template-service/src/main/java/com/liferay/layout/page/template/internal/upgrade/v3_4_4/LayoutPageTemplateStructureUpgradeProcess.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+			ServiceContext serviceContext = new ServiceContext();
+
+			while (resultSet.next()) {
+				_layoutLocalService.updateStatus(
+					resultSet.getLong("userId"), resultSet.getLong("plid"), 0,
+					serviceContext);
+
+				preparedStatement2.setLong(1, resultSet.getLong("plid"));
+				preparedStatement2.addBatch();
```
Comment (balazssk) [inline modules/apps/layout/layout-page-template-service/src/main/java/com/liferay/layout/page/template/internal/upgrade/v3_4_4/LayoutPageTemplateStructureUpgradeProcess.java:57]: Looks like there is a [rebuildLayoutPageTemplateStructure](https://github.com/liferay/liferay-portal/blob/master/modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/util/LayoutCopyHelperImpl.java#L176 ) call here, can you please check if the source Layout also has an unneeded LayoutPageTemplateStructure?

File: modules/apps/layout/layout-page-template-service/src/main/java/com/liferay/layout/page/template/internal/upgrade/v3_4_4/LayoutPageTemplateStructureUpgradeProcess.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import com.liferay.portal.kernel.model.LayoutConstants;
+import com.liferay.portal.kernel.service.LayoutLocalService;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Mark Gulacsy
```
Comment (balazssk) [inline modules/apps/layout/layout-page-template-service/src/main/java/com/liferay/layout/page/template/internal/upgrade/v3_4_4/LayoutPageTemplateStructureUpgradeProcess.java:27]: you can use Hungarian characters for @author

File: modules/apps/layout/layout-page-template-service/src/main/java/com/liferay/layout/page/template/internal/upgrade/v3_4_4/LayoutPageTemplateStructureUpgradeProcess.java (hunk 1 above)
Comment (markgulacsy) [reply modules/apps/layout/layout-page-template-service/src/main/java/com/liferay/layout/page/template/internal/upgrade/v3_4_4/LayoutPageTemplateStructureUpgradeProcess.java:39]: why do we need to use the service api to get the layouts?

Comment (balazssk) [reply modules/apps/layout/layout-page-template-service/src/main/java/com/liferay/layout/page/template/internal/upgrade/v3_4_4/LayoutPageTemplateStructureUpgradeProcess.java:39]: Layouts are part of the portal-impl module and we can only use SQL to query entities in the layout-page-template-service module in this UpgradeProcess [link](https://grow.liferay.com/share/DXP-+How+to+code+upgrade+process+and+backport+rules#section-DXP-+How+to+code+upgrade+process+and+backport+rules-Golden+rule)

**Comment by markgulacsy** [comment]
I will fix the upgradeProcess according to your review. I will open a new PR when it is ready. Thanks!

**Pull request closed without merging**